stats/stats.go

```diff
@@ -178,36 +178,42 @@ func readStats() {
 		case AUTH_REQ_IN:
 			logger.StatsLog.Infoln("Received Event: AUTH_REQ_IN: ", m)
 			t := popTrans(m.Id) // remove MSG in trans but use the time msg was received
-			m.T = t.T
-			ue := getUe(m.Supi)
-			ue.CReg.AuthReqInTime = m.T
-			x := m.T.Sub(ue.CReg.RegReqOutTime)
-			ue.CReg.RegReqAuthReq = x.Microseconds()
-			logger.StatsLog.Infoln("Time[us] between Reg Req & Auth Req ", ue.CReg.RegReqAuthReq)
+			if t != nil {
```

Comment:
Do you know why the transaction (Id) does not already exist in the `StatisticsEvent`?

Reply:
After further debugging, I see that the same transaction (Id=14) gets "popped" twice in the stats, which lead to `nil` pointer. Transaction 14 comes from `HandleInitialContextSetupRequest`

```
Received Event: PDU_SESS_ACC_IN, Id: 9
Received Event: UE_CTX_CMD_IN, Id: 12
Received Event: ICS_REQ_IN, Id: 14
Received Event: SVC_ACCEPT_IN, Id: 14
```

Reply:
Yes current design needs some changes to handle few corner cases and hence it crashes if transaction not found. I would say at least  gnbsim should not crash if no transaction found.